### PR TITLE
Fenwick tree implemented in Dart - #2173 resolved

### LINKED ISSUE #2173: Fenwick Tree
- [x] C
- [x] C++
- [ ] C# - @ritish099
- [ ] CoffeeScript
- [x] Dart - @Rukmini-Meda
- [ ] Go
- [x] Java
- [x] JS
- [x] Kotlin - @JasbirCodeSpace
- [x] Php - @goyalmuskan
- [x] Python
- [ ] Readme - @AJAYSHARMA2012
- [ ] Ruby
- [ ] Rust
- [ ] TypeScript

### PR DESCRIPTION
**Pull request by Rukmini-Meda**
Fixes #2173 

#### Checklist:
- [x] 4 space indentation.
- [x] Coding conventions are followed.
- [x] Input is taken dynamically.
- [x] Sample Input / Output is added at the end of file.
- [x] Logic Documentation (Comments).
- [x] File names are correct.

Make sure these boxes are checked before your pull request (PR) is ready to be reviewed and merged. Thanks!

#### Changes proposed in this pull request:
- Fenwick tree is implemented in Dart

#### Languages Used:
- Dart

#### Files Added:
- Fenwick_Tree.dart

> We're happy to help you get this ready -- don't be afraid to ask for help.

Thanks!

### CODE REVIEW

File: Fenwick_Tree/Fenwick_Tree.dart [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+void main(){
+
+    // Input of number of array elements
+    print("Enter number of elements in the array:");
+    var input = stdin.readLineSync();
+    int n = int.parse(input);
+
+    // Input of array elements
+    print("Enter array elements:");
+    input =stdin.readLineSync();
```
Comment (atarax665) [inline Fenwick_Tree/Fenwick_Tree.dart:74]: Give space after =

File: Fenwick_Tree/Fenwick_Tree.dart [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+                int result = findSum(tree_BIT,choices[0],choices[1]);
+                print(result);
+                break;
+        }
+    }
+}
+
+/**
+ * 
+ *   Sample Input and Output
```
Comment (atarax665) [inline Fenwick_Tree/Fenwick_Tree.dart:129]: Mention input and output separately.

**Comment by Rukmini-Meda** [comment]
@atarax665 I have made changes as suggested and squashed the commits. Please merge.

**Comment by Rukmini-Meda** [comment]
Please merge

**Comment by Rukmini-Meda** [comment]
@MastersAbh @SakshiShreya @singh-shreya6 @asha15 Please review and merge this PR.

File: Fenwick_Tree/Fenwick_Tree.dart [hunk 3]
```diff
@@ -0,0 +1,164 @@
...
+ * range queries optimally. It has O(n) space complexity,
+ * O(logn) time complexity to update, O(nlog(n)) time complexity to build
+ * and O(logn) time complexity to find sum in a given range.
+ * 
+ * References - Geeksforgeeks
+ * 
+ */
+
+
+// Importing required libraries
```
Comment (asha15) [inline Fenwick_Tree/Fenwick_Tree.dart:14]: please do the changes here as i asked you to do in other PR

File: Fenwick_Tree/Fenwick_Tree.dart [hunk 4]
```diff
@@ -0,0 +1,164 @@
...
+import 'dart:io';
+
```
Comment (asha15) [inline Fenwick_Tree/Fenwick_Tree.dart:16]: remove this line

File: Fenwick_Tree/Fenwick_Tree.dart (hunk 3 above)
Comment (Rukmini-Meda) [reply Fenwick_Tree/Fenwick_Tree.dart:14]: @asha15 I wrote in my own words in every PR I made. I only used GFG to learn and understand the algorithm. I hope you understand. But if you still think I should change, I will do it, please let me know.

File: Fenwick_Tree/Fenwick_Tree.dart (hunk 4 above)
Comment (Rukmini-Meda) [reply Fenwick_Tree/Fenwick_Tree.dart:14]: I removed that, committed and squashed, Please let me know about changes if any further and merge.

**Comment by Rukmini-Meda** [comment]
@asha15 I have removed references and squashed the commits. Please merge

**Pull request merged**